src/mac/MacDisplayInfo.mm

```diff
@@ -0,0 +1,60 @@
+#include "DisplayInfo.h"
+#include "UserInteractions.h"
+#include <CoreFoundation/CoreFoundation.h>
+#include <Cocoa/Cocoa.h>
+#include "vstgui/lib/platform/mac/cocoa/nsviewframe.h"
+
+namespace Surge
+{
+    namespace GUI
+    {
+        NSView *getViewFromFrame(CFrame *f)
+        {
+            IPlatformFrame *pf = f->getPlatformFrame();
+            VSTGUI::NSViewFrame *nf = dynamic_cast<VSTGUI::NSViewFrame*>(pf);
+            if (!nf)
+            {
+                Surge::UserInteractions::promptError("Platform frame is not an NSViewFrame. Please report this issue to developers.",
+                                                     "Software Error");
+                return NULL;
+            }
+            return nf->getNSView();
+        }
+        float getDisplayBackingScaleFactor(CFrame *f)
+        {
+            float defaultBackingScale = 2.0; // If I don't know, guess retina.
+            if (!f) return defaultBackingScale;
+            
+            NSView *v = getViewFromFrame(f);
+
+            // So objective c would run these but lets be explicit
+            if( !v || ![v window] || ![[v window] screen])
+            {
```

Comment:
Not sure why you want to have curly brackets here.

Reply:
I try to always use curly braces when I press return after an if.